lib/swaps/SwapClientsManager.ts

```diff
@@ -0,0 +1,150 @@
+  private repository: OrderBookRepository;
+
+  constructor(
+    private models: Models,
+    private config: Config,
+    private loggers: Loggers,
+    private pool: Pool,
+  ) {
+    this.repository = new OrderBookRepository(loggers.orderbook, models);
+  }
+
+  public async init() {
+    const initPromises = [];
+    // setup LND clients and initialize
+    for (const currency in this.config.lnd) {
+      const lndConfig = this.config.lnd[currency]!;
+      if (!lndConfig.disable) {
+        const lndClient = new LndClient(lndConfig, currency, this.loggers.lnd);
+        this.lndClients[currency] = lndClient;
+        this.swapClients.set(currency, lndClient);
+        initPromises.push(lndClient.init());
+      }
+    }
+    // setup raiden client and connect if configured
+    this.raidenClient = new RaidenClient(this.config.raiden, this.loggers.raiden);
+    initPromises.push(this.raidenClient.init());
+    await Promise.all(initPromises);
+    if (!this.raidenClient.isDisabled()) {
+      for (const currency of this.raidenClient.tokenAddresses.keys()) {
+        this.swapClients.set(currency, this.raidenClient);
+      }
+    }
+  }
+
+  public async add(currency: string, swapClient: SwapClient) {
+    if (swapClient === SwapClient.Raiden) {
+      this.swapClients.set(currency, this.raidenClient);
+      await this.setRaidenCurrencies();
+    } else if (swapClient === SwapClient.Lnd) {
+      let hasCurrency = false;
+      for (const lndCurrency in this.config.lnd) {
+        if (lndCurrency === currency) {
+          hasCurrency = true;
+        }
+      }
+      if (!hasCurrency) {
+        throw new Error(`lnd client for ${currency} is not configured`);
+      }
+    }
+  }
+
+  public async remove(currency: string) {
+    const swapClient = this.get(currency);
+    this.swapClients.delete(currency);
+    if (swapClient.type === SwapClient.Raiden) {
+      await this.setRaidenCurrencies();
+    }
+  }
+
+  public getLndPubKeys() {
+    const lndPubKeys: { [currency: string]: string | undefined } = {};
+    for (const currency in this.lndClients) {
+      lndPubKeys[currency] = this.lndClients[currency]!.pubKey;
+    }
+    return lndPubKeys;
+  }
+
+  public getRaidenAddress() {
+    return this.raidenClient.address;
+  }
+
+  public async getLndClientsInfo() {
+    const lnd: { [currency: string]: LndInfo | undefined } = {};
+    for (const currency in this.lndClients) {
+      const lndClient = this.lndClients[currency]!;
+      lnd[currency] = lndClient.isDisabled() ? undefined : await lndClient.getLndInfo();
+    }
+    return lnd;
+  }
+
+  public async getRaidenInfo() {
+    const raidenInfo = this.raidenClient.isDisabled() ? undefined : await this.raidenClient.getRaidenInfo();
+    return raidenInfo;
+  }
+
+  public get(currency: string): BaseClient {
+    const swapClient = this.swapClients.get(currency);
+    if (!swapClient) {
+      throw new Error(`swapClient for currency ${currency} not found`);
```

Comment:
Maybe this is where we can throw that error I was talking about in my last comment, and ideally make it a defined error type rather than a generic `Error` so we can easily check for it.